README.md

```diff
@@ -5,757 +5,864 @@
+|                   |
+|                   +-set()->
+|                   +-state: Car<{ designer: Person<{ name: 'Homer', age: 39 }> }, name: 'The Homer' }>
++-------------------+
 ```
 
-You can access transitions of composed states by using object property notation. Every time that you
-call a transition, you receive a new microstate and you start from the root of the original type.
+The property names are important when defining `class` types because they are used to reference composed microstates. You can use the object dot notation to access a composed microstate. Using the same example from above:
 
 ```js
-create(Person)
-  .age.increment()
-  .address.city.set('San Francisco')
-  .address.street.set('Market St').state;
-// => {
-//      name: 'Taras',
-//      age: 1,
-//      address: {
-//        street: 'Market St',
-//        number: 0,
-//        city: 'San Francisco'
-//      }
-//    }
+theHomerCar.designer.state;
+//> Person<{ name: 'Homer', age: 39 }>
+
+theHomerCar.designer.age.state;
+//> 39
+
+theHomerCar.name.state;
+//> The Homer
 ```
 
-The objects can be of any complexity and can even support recursion.
+## Array Microstates
+
+Quite often it is helpful to describe your data as a collection of types. For example, a blog might have an array of posts. To do this, you can use the array of type notation, `[Post]`. This signals that Microstates of this type represent an array whose members are each of the `Post` type.
 
 ```js
-class Person {
-  name = String;
-  father = Person;
+class Blog {
+  posts = [Post];
+}
+
+class Post {
+  id = Number;
+  title = String;
 }
 
-create(Person, { name: 'Stewie' })
-  .father.name.set('Peter')
-  .father.father.name.set('Mr Griffin')
-  .father.father.father.name.set('Mr Giffin Senior').state;
-// => { name: 'Stewie',
-//      father: {
-//        name: 'Peter',
-//        father: {
-//          name: 'Mr Griffin,
-//          father: {
-//            name: 'Mr Griffin Senior'
-//          }
-//        }
-//      }
-//    }
+let blog = create(Blog, {
+  posts: [
+    { id: 1, title: "Hello World" },
+    { id: 2, title: "Most fascinating blog in the world" }
+  ]
+});
+
+blog.posts[0].state;
+//> Post<{ id: 1, title: 'Hello World' }>
+
+blog.posts[1].state;
+//> Post<{ id: 2, title: 'Most fascinating blog in the world' }>
 ```
 
-We currently do not support composition inside of arrays but we're looking into supporting it in the
-future.
+When you're working with parameterized types, the shape of the Microstate is determined by the value. In this case, `posts` is created with two items which will, in turn, create a Microstate with two items. Each item will be a Microstate of type `Post`. If you push another item onto the `posts` Microstate, it'll be treated as a `Post`.
+
+```js
+let blog2 = blog.posts.push({ id: 3, title: "It is only getter better" });
+
+blog2.posts[2].state;
+//> Post<{ id: 3, title: 'It is only getter better' }>
+```
```

Comment:
It's worth calling out at the end of this section, something like

> Notice how we didn't have to do _any_ extra work to define the state transition of adding another post to the list? That's the power of composition!

Or something like that. The awesomeness of microstates is that you don't even notice how awesome it is. Folks might read that and take `ArrayType#push` for granted because "obvs, right?" No. Not obvs. That was true ultimate power you just witnessed.

Reply:
> Or something like that. The awesomeness of microstates is that you don't even notice how awesome it is. Folks might read that and take ArrayType#push for granted because "obvs, right?" No. Not obvs. That was true ultimate power you just witnessed.

This is blog post material